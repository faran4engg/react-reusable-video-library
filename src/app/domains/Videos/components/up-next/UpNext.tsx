import { FC } from 'react';
import { SvgIcon } from 'app/domains/Common/components/svg-icon';
import { UpNextProps } from './types';
import { useHistory } from 'react-router-dom';

const UpNext: FC<UpNextProps> = ({ approvedVideos, category }) => {
  const history = useHistory();
  const list = Object.keys(approvedVideos)
    .map(key => ({ ...approvedVideos?.[key], id: key }))
    .filter(vid => vid.category === category);

  return (
    <div className="col-span-1">
      <h2 className="flex justify-between my-4 md:my-1">
        ‚ñ∂Ô∏è Up Next
        <span>scroll üëá</span>
      </h2>

      <div className="flex flex-wrap mb-4 overflow-y-auto h-96 md:h-80 lg:h-96">
        <ul className="flex flex-col w-full">
          {list.map(video => (
            <li
              key={video.timestamp}
              className="flex flex-row mb-2 border-gray-400"
            >
              <div className="flex items-center flex-1 p-3 overflow-hidden rounded-md shadow-md cursor-pointer select-none bg-light-mode dark:bg-dark-mode">
                <div className="flex flex-col items-center justify-center mr-2 rounded-md ">
                  <div className="w-8 h-8">
                    <SvgIcon icon="PlayListVideIcon" />
                  </div>
                </div>

                <div
                  className="flex flex-col overflow-hidden font-semibold text-gray-600 dark:text-gray-400"
                  onClick={() => history.push(`/watch/${video.id}`)}
                >
                  <p className="truncate text-md">{video.title}</p>
                </div>
              </div>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default UpNext;
