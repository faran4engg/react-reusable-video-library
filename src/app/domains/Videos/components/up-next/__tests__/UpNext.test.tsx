import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import UpNext from '../UpNext';

const renderer = createRenderer();

describe('<UpNext />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<UpNext />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
