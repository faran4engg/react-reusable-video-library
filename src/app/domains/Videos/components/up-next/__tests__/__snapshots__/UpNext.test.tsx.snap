// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpNext /> should render and match the snapshot 1`] = `
<div
  className="col-span-1"
>
  <SectionTitle>
    ▶️ Up Next
  </SectionTitle>
  <div
    className="flex flex-wrap mb-4"
  >
    <ul
      className="flex flex-col w-full "
    >
      <li
        className="flex flex-row mb-2 border-gray-400"
      >
        <div
          className="flex flex-1 items-center p-4 rounded-md shadow-md transition duration-500 ease-in-out transform cursor-pointer select-none  hover:-translate-x-1 bg-light-mode dark:bg-dark-mode"
        >
          <div
            className="flex flex-col justify-center items-center mr-2 w-10 h-10 rounded-md"
          >
            <SvgIcon
              icon="PlayListVideIcon"
            />
          </div>
          <div
            className="flex-1"
          >
            <div
              className="font-semibold text-gray-600 text-md dark:text-gray-400"
            >
              sample title
            </div>
          </div>
        </div>
      </li>
    </ul>
    <ul
      className="flex flex-col w-full "
    >
      <li
        className="flex flex-row mb-2 border-gray-400"
      >
        <div
          className="flex flex-1 items-center p-4 rounded-md shadow-md transition duration-500 ease-in-out transform cursor-pointer select-none  hover:-translate-x-1 bg-light-mode dark:bg-dark-mode"
        >
          <div
            className="flex flex-col justify-center items-center mr-2 w-10 h-10 rounded-md"
          >
            <SvgIcon
              icon="PlayListVideIcon"
            />
          </div>
          <div
            className="flex-1"
          >
            <div
              className="font-semibold text-gray-600 text-md dark:text-gray-400"
            >
              sample title
            </div>
          </div>
        </div>
      </li>
    </ul>
    <ul
      className="flex flex-col w-full "
    >
      <li
        className="flex flex-row mb-2 border-gray-400"
      >
        <div
          className="flex flex-1 items-center p-4 rounded-md shadow-md transition duration-500 ease-in-out transform cursor-pointer select-none  hover:-translate-x-1 bg-light-mode dark:bg-dark-mode"
        >
          <div
            className="flex flex-col justify-center items-center mr-2 w-10 h-10 rounded-md"
          >
            <SvgIcon
              icon="PlayListVideIcon"
            />
          </div>
          <div
            className="flex-1"
          >
            <div
              className="font-semibold text-gray-600 text-md dark:text-gray-400"
            >
              sample title
            </div>
          </div>
        </div>
      </li>
    </ul>
    <ul
      className="flex flex-col w-full "
    >
      <li
        className="flex flex-row mb-2 border-gray-400"
      >
        <div
          className="flex flex-1 items-center p-4 rounded-md shadow-md transition duration-500 ease-in-out transform cursor-pointer select-none  hover:-translate-x-1 bg-light-mode dark:bg-dark-mode"
        >
          <div
            className="flex flex-col justify-center items-center mr-2 w-10 h-10 rounded-md"
          >
            <SvgIcon
              icon="PlayListVideIcon"
            />
          </div>
          <div
            className="flex-1"
          >
            <div
              className="font-semibold text-gray-600 text-md dark:text-gray-400"
            >
              sample title
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
